File: main/src/main/java/com/JewelleryStore/main/model/Product.java
package com.JewelleryStore.main.model;

import com.JewelleryStore.main.model.others.ProductSerializer;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "products")
@JsonSerialize(using = ProductSerializer.class)
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "product_id", updatable = false, nullable = false)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private double price;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String scale;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private ProductCategory category;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<ProductImage> images;
}

File: FigureStore/src/models/Product.ts
interface Product {
    id: string;
    name: string;
    description: string;
    price: number;
    brand: string;
    scale: string;
    category: Category[]
    images: Image[];
}

interface Image {
    images: {
        id: string;
        url: string
    }
}

interface ProductDTO {
  name: string;
  description: string;
  price: number;
  brand: string;
  scale: string;
  category: string;
}
